### DISCUSSION TITLE
考虑出一个LTS版本可以吗？

### DISCUSSION THREAD

**Discussion by yottayuan** [dotnetcore/BootstrapBlazor#3259] [Q&A] [answered]
大佬，您们的开发太勤奋，太猛了，新特性的快速添加同时伴随着新BUG引入的可能性。我们在生产环境下引用咱们的库，不知道选哪个版本合适啊，能不能为生产环境准备一个LTS版本？这个版本专为生产环境准备，每次升级只修复Bug，不引入新特性？

**Comment by bb-auto** [comment]
@yottayuan Thank you for reporting. We will give triage later.

**Comment by densen2014** [comment]
应该没有这个计划。 怎么你实际上碰到了很多bug吗？有没有故事可以分享一下

**Reply by yottayuan** [reply]
实际碰到的Bug倒不是很多，就是目前咱们这种快速迭代的方式让我们很没安全感（主要是我们项目实施周期也很紧张，很难保证测试到位），我们在项目实施中没有使用nuget包，而是引入咱们工程的源码，这样方便我们学习咱们工程，同时方便在线Debug。

**Comment by ArgoZhang** [comment]
1. 首先已投入生产环境的可以不升级
2. 针对当前遇到 bug 可以单独报提供单独补丁包

技术实验性探索肯定是最新版最好

组件单元测试代码覆盖率 100% 但是确实仍然不能排除有新 BUG 出现，基本都是逻辑未考虑或者多个参数叠加导致，这个是所有开发面临的问题，并不是我们组件库特有的

**Reply by yottayuan** [reply]
我们自己在使用咱们项目的源码内部维护一个版本，但对咱们项目的理解肯定不深，对这个版本的升级也是隔些时间使用最新的源码覆盖一下，说实话，也没安全感，不确定最新的源码可能存在的问题。语言组织可能有些混乱，就是想表达，如果咱们官方提供一个类似半年或一年升级一次的LTS，维护周期1年或2年的LTS最好了，毕竟，官方最了解自己的工程。

**Comment by yottayuan** [comment]
这样的好处，我们在生产环境中使用上一个LTS版本，然后在开发环境中使用当前的LTS版本，上一个LTS版本因为经过长时间的BUG修复，非常稳定，开发的时候使用当前的LTS版本，方便发现当前LTS版本的BUG来提交到您们这里。否则可能需要一直更新咱们的最新版本。这样我们可以在生产和开发环境下和咱们的LTS版本保持一致。

**Comment by j4587698** [comment]
没有必要啊，新特性你不用不就跟旧的一样嘛？除非有破坏性更新。现在破坏性更新很少了，绝大部分情况下都可以无缝升级啊

**Reply by yottayuan** [reply]
我们内部在维护一个内部框架工程，BootsstrapBlazor是其中的一个核心库，我们是希望我们这个框架跟随BootstrapBlazor一起进步，逐步使用新特性的。比如前段时间，升级版本后，我们也没有引入新特性，Table报异常，现在版本应该是修复了（我还没来得及更新，我看更新日志上有几个关于Table的修复），我是有些担心，升级版本后，虽然没引用新特性，原来的特性因为新特性的引入出问题了。

**Reply by ArgoZhang** [reply, marked as answer]
你的意思我明白了。这种公司都是跟我有合作的，免费的这种方式您换位思考一下，会不会累死我们？

**Reply by ArgoZhang** [reply]
@yottayuan 加一下我微信 5196060
